### PR TITLE
Closes #3304: Add Quickstart Exposed Filter options to existing views

### LINKED ISSUE #3304: Add Quickstart Exposed Filter view displays
## Motivation
I want default views available to use the Quickstart Exposed Filters functionality.

## Proposed Resolution
Add new view displays to all content type views that uses Quickstart Exposed Filters.
- AZ News
- AZ Pages
- AZ Events
- AZ People

### Alternative solutions
Consider modifying existing view displays to use the filter block if placed.

### PR DESCRIPTION
**Pull request by danahertzberg**
## Description
This modifies existing views to have the ability to work with filters that are exposed in an external block. 

## Related issues
**Blocked by #2998** 
Need to verify CSS compatibility before merging this.

## Remaining to do items
- [x] Update views to use the options for the term filters (Quickstart Exposed Filters in the filter settings per field)
- [ ] Investigate reset filters button - seems to work on every view except AZ Person

## How to test
1. Create a page with any of the following views
   - AZ Person 
      - Grid page
      - Grid - Modal
      - Row page
      - Grid 3 col
      - Grid - Modal 3 col
   - AZ Events
      - Grid view
      - Grid view with images
      - Past grid view with images
   - AZ News
      - Paged row view 
      - Grid view
   - AZ Pages
      - Row page 
      - Grid page
3. Add the associated exposed filter block to that page
4. Ensure the filter functions as expected

### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
   - [ ] Bug fix
   - [ ] Accessibility, performance, or security improvement
   - [ ] Critical institutional link or brand change
   - [ ] Adding experimental module
   - [ ] Update experimental module
- **Minor release changes**
   - [x] New feature
   - [ ] Breaking or visual change to existing behavior
   - [ ] Upgrade experimental module to stable
   - [ ] Enable existing module by default or database update
   - [ ] Non-critical brand change
   - [ ] New internal API or API improvement with backwards compatibility
   - [ ] Risky or disruptive cleanup to comply with coding standards
   - [ ] High-risk or disruptive change (requires upgrade path, risks regression, etc.)
- **Other or unknown**
   - [ ] Other or unknown

### Drupal core
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch level release (non-security bug-fix release)
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major or minor level update
- **Other or unknown**
   - [ ] Other or unknown

### Drupal contrib projects
- **Patch release changes**
   - [ ] Security update
   - [ ] Patch or minor level update
   - [ ] Add new module
   - [ ] Patch removal that's no longer necessary
- **Minor release changes**
   - [ ] Major level update
- **Other or unknown**
   - [ ] Other or unknown

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by danahertzberg** [comment]
@tadean Yes, it is. What process effects does that have? Or is more of the enabling by default as the issue?

**Comment by tadean** [comment]
> @tadean Yes, it is. What process effects does that have? Or is more of the enabling by default as the issue?

Correct, the enabling by default. So far all of our experimental modules are opt-in, I think.

File: az_quickstart.install [hunk 1]
```diff
@@ -195,3 +195,10 @@ function az_quickstart_update_1020802() {
 function az_quickstart_update_1020803() {
   \Drupal::service('module_installer')->uninstall(['media_library_theme_reset']);
 }
+
+/**
+ * Enable az_finder module by default.
+ */
+function az_quickstart_update_1020804() {
```
Comment (joeparsons) [inline az_quickstart.install:202]:
```suggestion
function az_quickstart_update_1021001() {
```

If this is meant to be a 2.10.x-only change, we should change this number.

**Comment by danahertzberg** [comment]
> Correct, the enabling by default. So far all of our experimental modules are opt-in, I think.

Can this module be a dependency of other modules enabled by default if it's not enabled itself?

**Comment by tadean** [comment]
> > Correct, the enabling by default. So far all of our experimental modules are opt-in, I think.
> 
> Can this module be a dependency of other modules enabled by default if it's not enabled itself?

Not as far as I'm aware. It might be possible to implement that in `az_finder` by having it change the existing views by altering the view to use the `az_better_exposed_filters` exposed form plugin when it's enabled.

**Comment by danahertzberg** [comment]
## Notes
Add quickstart config in the finder module
When experimental module is enabled, then it alters the view
Hooks to change the view before it executes as alternative approach
Look at having stable functionality in 2.11, then require these changes at install

## To do
- [ ] config/quickstart folder in az_finder - move copies of view files here

**Review by joeparsons — APPROVED** [review]

**Review by trackleft — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**